base/tuple.jl

```diff
@@ -16,8 +16,15 @@ NTuple
 
 # convenience function for extracting N from a Tuple (if defined)
 # else return `nothing` for anything else given (such as Vararg or other non-sized Union)
-_counttuple(::Type{<:NTuple{N,Any}}) where {N} = N
-_counttuple(::Type) = nothing
+
+"""arity(::Type{<:NTuple})
+Function
+Returns the number of elements in the tuple type.
+Returns nothing for anything else.
+"""
+arity(::Type{<:NTuple{N,Any}}) where {N} = N
+arity(::Type) = nothing
+const _counttuple = arity
```

Comment:
Now that I think more about it, given the leading `_`, I think it's clear that this was not part of our public API and you should be safe to just delete this.

Also, i checked here, and it doesn't seem like anyone is currently using it:
https://juliahub.com/ui/CodeSearch?q=_counttuple&u=all&t=all

Reply:
Alias removed.